crates/rustc_codegen_spirv/src/builder/spirv_asm.rs

```diff
@@ -801,7 +801,17 @@ impl<'cx, 'tcx> Builder<'cx, 'tcx> {
 
             (OperandKind::ImageOperands, Some(word)) => {
                 match parse_bitflags_operand(IMAGE_OPERANDS, word) {
-                    Some(x) => inst.operands.push(dr::Operand::ImageOperands(x)),
+                    Some(x) => {
+                        inst.operands.push(dr::Operand::ImageOperands(x));
+                        for _ in 0..x.bits().count_ones() {
```

Comment:
Is this correct? For example, the `Grad` image operand is a single bit (0x4) but it takes two operands, and there are also a handful that have zero operands.

Reply:
good catch! I need to adjust `rspirv` as well.